### PR TITLE
Made the calculation of the asset_loss_table optional

### PR DESCRIPTION
**Pull request by micheles**
In this way we can save a lot of memory and data transfer, see https://github.com/gem/oq-risklib/issues/711. The demos are running here: https://ci.openquake.org/job/zdevel_oq-risklib/1397/

### CODE REVIEW

File: openquake/calculators/event_based_risk.py [hunk 1]
```diff
@@ -34,17 +34,30 @@
 F32 = numpy.float32
 
 
+def build_el_dtypes(insured_losses):
+    """
+    :param bool insured_losses: job.ini configuration parameter
+    :returns: ela_dt and elt_dt
+    """
+    ela_list = [('rup_id', U32), ('ass_id', U32), ('loss', F32)]
```
Comment (ptormene) [inline openquake/calculators/event_based_risk.py]: Please add a comment describing ela and elt

File: openquake/calculators/event_based_risk.py [hunk 2]
```diff
@@ -34,17 +34,30 @@
...
-    :param r_data:
-        a list of pairs `(r, data)` where `r` is a realization index and `data`
-        is an array of pairs `(rupture_id, loss)` where loss is an array of
-        shape L (in absence of insured losses) or L * 2 (with insured losses)
+    :param lr_data:
+        a list of triples `(l, r, data)` where `l` is the loss type index,
+        `r` is the realization index and `data` is an array of kind
+        `(rupture_id, loss)` or `(rupture_id, loss, loss_ins)`
     :param insured_losses:
         job.ini configuration parameter
```
Comment (ptormene) [inline openquake/calculators/event_based_risk.py]: you might specify also in this docstring that it's a bool

File: openquake/qa_tests_data/event_based_risk/case_2/expected/agg_losses-b1,b1-structural.csv [hunk 3]
```diff
@@ -1,27 +1,26 @@
...
-col=00~ses=0014~src=3~rup=15-01,1.153860E+03
-col=00~ses=0015~src=2~rup=8-01,8.886328E+01
-col=00~ses=0015~src=3~rup=13-01,4.398730E+02
-col=00~ses=0016~src=3~rup=13-01,1.250175E+03
-col=00~ses=0017~src=1~rup=1-01,3.473117E+02
-col=00~ses=0017~src=2~rup=7-01,1.422718E+02
-col=00~ses=0019~src=2~rup=7-01,5.276436E+02
-col=00~ses=0020~src=1~rup=1-01,4.899049E+02
-col=00~ses=0020~src=2~rup=7-01,1.815601E+02
+0,1.891084E+02
```
Comment (ptormene) [inline openquake/qa_tests_data/event_based_risk/case_2/expected/agg_losses-b1,b1-structural.csv]: Is it ok to have this without a header?

**Comment by ptormene** [comment]
Besides minor comments, LGTM

File: openquake/calculators/event_based_risk.py (hunk 1 above)
Comment (micheles) [inline openquake/calculators/event_based_risk.py]: Done

File: openquake/calculators/event_based_risk.py (hunk 2 above)
Comment (micheles) [inline openquake/calculators/event_based_risk.py]: Done

**Pull request merged**